devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -1378,21 +1414,43 @@ bool HumanStateProvider::impl::applyRpcCommand()
         std::iota(jointZeroIndices.begin(), jointZeroIndices.end(), 0);
 
         // Compute secondary calibration for the selected links setting to zero the given joints
-        computeSecondaryCalibrationRotationsForChain(jointZeroIndices, baseTransformSolution, linkToCalibrateIndices);
+        computeSecondaryCalibrationRotationsForChain(jointZeroIndices, iDynTree::Transform::Identity(), linkToCalibrateIndices, "");
+        break;
+    }
+    case rpcCommand::calibrateAllWithWorld: {
+        // Check if the chose baseLink exist in the model
+        std::string baseLinkName = commandPro->baseLinkName;
+        if(!(kinDynComputations->getRobotModel().isFrameNameUsed(baseLinkName)))
+        {
+            yWarning() << LogPrefix << "link " << baseLinkName << " choosen as base for secondaty calibration is not valid";
+            return false;
+        }
+
+        // Select all the links and the joints
+        // add all the links of the model to [linkToCalibrateIndices]
+        linkToCalibrateIndices.resize(kinDynComputations->getNrOfLinks());
+        std::iota(linkToCalibrateIndices.begin(), linkToCalibrateIndices.end(), 0);
+
+        // add all the joints of the model to [jointZeroIndices]
+        jointZeroIndices.resize(kinDynComputations->getNrOfDegreesOfFreedom());
+        std::iota(jointZeroIndices.begin(), jointZeroIndices.end(), 0);
```

Comment:
Nice, thanks! Didn't know about `iota` before.

Reply:
me neither, I just found it on the web :grimacing: